docs/sync-overview.md

```diff
@@ -0,0 +1,57 @@
+# Sync Overview
+
+This document provides a high-level overview of how syncing works.  **Note**: each component has its own quirks and will handle sync slightly differently than the general process described here.
+
+## General flow and architecture
+
+- **Crates involved**:
+  - The `sync15` and `support/sync15-traits` handle the general syncing logic and define the `SyncEngine` trait
+  - `SyncEngine` is implemented by each component crate (login, places, autofill, etc).
```

Comment:
super nit: I think this can be a secondary point under the first point since `SyncEngine` is not a crate

Reply:
I was thinking of the login/places/autofill crates, but it wasn't very clear.  I'll try to reverse it, maybe that will help.